runtime/interpreter/sharedstate.go

```diff
@@ -49,6 +51,7 @@ func NewSharedState(config *Config) *SharedState {
 		},
 		inStorageIteration:             false,
 		storageMutatedDuringIteration:  false,
+		inAttachmentIteration:          map[*CompositeValue]bool{},
```

Comment:
Maybe initialize this lazily, so we do not have the pay the cost of allocating it even when the program involves no attachments.